server/svix-server/src/queue/redis.rs

```diff
@@ -51,10 +55,17 @@ pub async fn new_pair(pool: RedisPool) -> (TaskQueueProducer, TaskQueueConsumer)
         let batch_size: isize = 50;
         let task_validity_duration = chrono::Duration::from_std(TASK_VALIDITY_DURATION).unwrap();
 
-        let pool = worker_pool;
-        loop {
+        {
+            let pool = worker_pool.clone();
             let mut pool = pool.get().await.unwrap();
 
+            // drain legacy queues:
+            migrate_legacy_queues(&mut pool).await;
+        }
+
+        loop {
+            let mut pool = worker_pool.get().await.unwrap();
```

Comment:
Why are we getting the pool here and didn't do before?

Reply:
This is the code on `main` currently:
```
    let worker_pool = pool.clone();
    tokio::spawn(async move {
        // FIXME: enforce we only have one such worker (locking)
        let batch_size: isize = 50;
        let task_validity_duration = chrono::Duration::from_std(TASK_VALIDITY_DURATION).unwrap();

        let pool = worker_pool;
        loop {
            let mut pool = pool.get().await.unwrap();
```
I removed the (what I think is redundant) `let pool = worker_pool` assignment.

Reply:
Ahh missed that, cool.